worldengine/draw.py

```diff
@@ -406,6 +406,113 @@ def draw_biome(world, target):
             v = biome[y, x]
             target.set_pixel(x, y, _biome_colors[v])
 
+def draw_scatter_plot(world, size, target):
+    """ This function can be used on a generic canvas (either an image to save
+        on disk or a canvas part of a GUI)
+    """
+    min_humidity = None
+    max_humidity = None
+    min_temperature = None
+    max_temperature = None
+    for y in range(world.height):
+        for x in range(world.width):
+            if world.is_land((x, y)):
+                t = world.temperature_at((x, y))
+                p = world.humidity['data'][y, x]
+                if min_temperature is None or t < min_temperature:
+                    min_temperature = t
+                if max_temperature is None or t > max_temperature:
+                    max_temperature = t
+                if min_humidity is None or p < min_humidity:
+                    min_humidity = p
+                if max_humidity is None or p > max_humidity:
+                    max_humidity = p
+    
+    temperature_delta = max_temperature - min_temperature
+    humidity_delta = max_humidity - min_humidity
+    
+    for y in range(0, size):
```

Comment:
Would you mind adding comments to every step of the preparation? Just two words to make the code more digestible. Somebody will come after you at some point. ;)